prov/rxd/src/rxd_av.c

```diff
@@ -415,8 +418,11 @@ int rxd_av_create(struct fid_domain *domain_fid, struct fi_av_attr *attr,
 err2:
 	ofi_av_close(&av->util_av);
 err1:
-	free(av->fi_addr_table);
-	free(av->rxd_addr_table);
-	free(av);
+	if (av->fi_addr_table)
+		free(av->fi_addr_table);
+	if (av->rxd_addr_table)
+		free(av->rxd_addr_table);
+	if (av)
+		free(av);
```

Comment:
free can handle null pointers.  A bigger concern is that there's a check for av == null, which implies that trying to access av->anything will lead to a crash